lib/beaker-puppet/install_utils/puppet5.rb

```diff
@@ -0,0 +1,109 @@
+        file_local_path = fetch_http_file(file_folder, filename, dst_folder)
+        file_hash = YAML.load_file(file_local_path)
+
+        # getting the relevant data from the file
+        logger.perf_output("YAML HASH BELOW:")
+        logger.perf_output(file_hash)
+        # pp file_hash
+        logger.perf_output("PLATFORM_DATA BELOW:")
+        logger.perf_output(file_hash[:platform_data])
+        return buildserver_projectsha_url, file_hash
+      end
+
+      # get the build artifact URL for a host
+      #
+      # @param [Host] host Host to get artifact URL for
+      # @param [Hash] build_details Details of the build in a hash
+      # @param [String] build_url URL to the build
+      #
+      # @return [String] URL to the build artifact
+      def hosts_artifact_url(host, build_details, build_url)
+        platform_key = host[:platform]
+        variant, version, arch, _ = platform_key.to_array
+        # step: map platform strings into platform_key's for the hash
+        if variant == 'debian' && arch == 'x86_64'
+          platform_key.sub!('x86_64', 'amd64')
+        elsif variant == 'fedora'
+          platform_key.sub!(version, "f#{version}")
+        end
+
+        logger.debug("PLATFORM SPECIFIC INFO for #{host} platform '#{platform_key}':")
+        logger.debug(" - #{build_details[:platform_data][platform_key]}, isnil? #{build_details[:platform_data][platform_key].nil?}")
+        artifact_buildserver_path = build_details[:platform_data][platform_key][:artifact]
+        fail_test('no artifact_buildserver_path found') if artifact_buildserver_path.nil?
+        artifact_url = "#{build_url}/#{artifact_buildserver_path}"
+
+        logger.debug("- artifact url: #{artifact_url}")
+        artifact_url_correct = link_exists?( artifact_url )
+        fail_test('artifact url built incorrectly') if !artifact_url_correct
+        artifact_url
+      end
+
+      # install build artifact on the given host
+      #
+      # @param [Host] host Host to install artifact on
+      # @param [String] project_name Name of project for artifact. Needed for OSX installs
+      # @param [String] artifact_url URL of the project install artifact
+      #
+      # @return nil
+      def install_artifact_on(host, artifact_url, project_name)
+        variant, _, _, _ = host[:platform].to_array
+        onhost_package_file = nil
+        if variant == 'solaris'
+          artifact_filename = File.basename( artifact_url )
+          artifact_folder = File.dirname( artifact_url )
+          fetch_http_file( artifact_folder, artifact_filename, '.' )
+          scp_to host, artifact_filename, '/tmp'
+          onhost_package_file = "/tmp/#{ artifact_filename }"
+        elsif variant == 'debian'
+          onhost_package_file = '/tmp/puppet.deb'
+          on host, "wget -O #{ onhost_package_file } #{ artifact_url }"
+        elsif variant == 'osx'
+          on host, "curl -O #{ artifact_url }"
+          onhost_package_file = "#{ project_name }*"
```

Comment:
hm, I'm not seeing a method that abstracts between these two right off the bat.

There's a `fetch_http_file` method that we often use with `scp_to` to install PE artifacts ([`install_repo_configs` example](https://github.com/puppetlabs/beaker-puppet/blob/master/lib/beaker-puppet/install_utils/foss_utils.rb#L966-L975)), but if we can get them right to the SUT rather than having to go through the coordinator that would be more efficient.

This pattern is pretty typical in the FOSS installs (just straight using `on` to execute curl or wget), but we don't typically see it side-by-side because of the way the work is split up in the install code (historically, we've done _everything_ for a platform in one body of a switch statement, rather than break up the tasks of download artifact vs install artifact). Refactoring this work into a helper method seems like a useful thing, but something that can be done separately from this work.

@joshcooper am I just missing the method, or do you think creating that helper method should block this PR?